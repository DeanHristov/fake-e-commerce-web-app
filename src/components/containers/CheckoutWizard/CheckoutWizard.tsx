'use client';

import { FC, useState } from 'react';
import { ArrowRightIcon } from '@heroicons/react/24/solid';

import WizardContent from './WizardContent/WizardContent';
import WizardHeading from './WizardHeading/WizardHeading';
import Widget from '@/components/containers/Widget';
import Receipt from '@/components/ui/Receipt';
import { useSelector } from '@/store';
import { selectShoppingCart } from '@/store/slices';
import { Utils } from '@/utils/Utils';
import Button from '@/components/ui/Button';
import { ICart } from '@/types';

export enum ACTIVE_VARIANT {
  PREVIEW = 1,
  RECEIPT = 3,
}

export interface ICheckoutWizardProps {}

// TODO Add step history to make possible the user go back through the steps
const CheckoutWizard: FC<ICheckoutWizardProps> = () => {
  // const dispatch = useDispatch();
  const shoppingCart: ICart = useSelector(selectShoppingCart);
  const shouldDisable: boolean = Utils.isEmpty(shoppingCart.products);
  const [activeTab, setActiveTab] = useState<ACTIVE_VARIANT>(
    ACTIVE_VARIANT.PREVIEW,
  );

  return (
    <Widget
      title={'Check out'}
      className="w-full checkout-container min-h-full space-y-4"
    >
      <div className="tab-heading h-12 flex justify-center items-center">
        <WizardHeading
          variant={ACTIVE_VARIANT.PREVIEW}
          onClick={setActiveTab}
          active={activeTab}
        />
        <WizardHeading
          disabled={shouldDisable}
          variant={ACTIVE_VARIANT.RECEIPT}
          onClick={setActiveTab}
          active={activeTab}
        />
      </div>
      <div className="tab-content h-full flex">
        <WizardContent isActive={activeTab === 1}>
          <h3 className="text-center">
            {"Imagine you've entered your card data!"}
          </h3>
          <Button
            disabled={shouldDisable}
            onClick={() => setActiveTab(ACTIVE_VARIANT.RECEIPT)}
            rightIcon={<ArrowRightIcon className="w-8 h-8" />}
            title={'Pay & Go'}
            className="mt-10"
          />
        </WizardContent>

        <WizardContent isActive={activeTab === 3}>
          <div className="max-w-xs h-auto mx-auto shadow-md">
            <Receipt
              currency={shoppingCart.currency || 'EUR'}
              total={shoppingCart.discountedTotal}
              products={Object.values(shoppingCart.products)}
            />
          </div>
        </WizardContent>
      </div>
    </Widget>
  );
};

export default CheckoutWizard;
