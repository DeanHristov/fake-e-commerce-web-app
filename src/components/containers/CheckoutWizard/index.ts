import CheckoutWizard, {
  ACTIVE_VARIANT,
  ICheckoutWizardProps,
} from './CheckoutWizard';
import WizardHeading, {
  IWizardHeadingProps,
} from './WizardHeading/WizardHeading';
import WizardContent, {
  IWizardContentProps,
} from './WizardContent/WizardContent';

export type {
  ICheckoutWizardProps,
  IWizardHeadingProps,
  IWizardContentProps,
  ACTIVE_VARIANT,
};

export { WizardHeading, WizardContent };
export default CheckoutWizard;
