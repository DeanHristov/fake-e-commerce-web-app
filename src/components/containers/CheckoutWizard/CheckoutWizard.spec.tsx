import { render } from '@testing-library/react';
import CheckoutWizard from './';

describe('Containers/Controls <CheckoutWizard {...} />', () => {
  it('Should be able to render the default state of the Component', () => {
    const { container } = render(<CheckoutWizard />);

    expect(container).toMatchSnapshot();
  });
});
