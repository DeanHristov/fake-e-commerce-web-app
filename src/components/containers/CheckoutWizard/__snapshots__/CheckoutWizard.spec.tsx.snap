// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers/Controls <CheckoutWizard {...} /> Should be able to render the default state of the Component 1`] = `
<div>
  <div
    class="bg-white border border-gray-300 rounded-lg p-4 shadow-md inline-block relative w-full checkout-container min-h-full space-y-4"
  >
    <h3
      class="text-base text-black font-bold pb-2"
    >
      Check out
    </h3>
    <div
      class="tab-heading h-12 flex justify-center items-center"
    >
      <div
        class="wizard-heading border-green-500"
      >
        <div
          class="wizard-heading-circle border-green-500 active"
          data-testid="circle-button"
        >
          <svg
            aria-hidden="true"
            class="wizard-heading-icon text-green-500"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.5 6v.75H5.513c-.96 0-1.764.724-1.865 1.679l-1.263 12A1.875 1.875 0 004.25 22.5h15.5a1.875 1.875 0 001.865-2.071l-1.263-12a1.875 1.875 0 00-1.865-1.679H16.5V6a4.5 4.5 0 10-9 0zM12 3a3 3 0 00-3 3v.75h6V6a3 3 0 00-3-3zm-3 8.25a3 3 0 106 0v-.75a.75.75 0 011.5 0v.75a4.5 4.5 0 11-9 0v-.75a.75.75 0 011.5 0v.75z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
      <div
        class="wizard-heading border-gray-300"
      >
        <div
          class="wizard-heading-circle border-gray-300"
          data-testid="circle-button"
        >
          <svg
            aria-hidden="true"
            class="wizard-heading-icon text-gray-300"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.731 2.269a2.625 2.625 0 00-3.712 0l-1.157 1.157 3.712 3.712 1.157-1.157a2.625 2.625 0 000-3.712zM19.513 8.199l-3.712-3.712-8.4 8.4a5.25 5.25 0 00-1.32 2.214l-.8 2.685a.75.75 0 00.933.933l2.685-.8a5.25 5.25 0 002.214-1.32l8.4-8.4z"
            />
            <path
              d="M5.25 5.25a3 3 0 00-3 3v10.5a3 3 0 003 3h10.5a3 3 0 003-3V13.5a.75.75 0 00-1.5 0v5.25a1.5 1.5 0 01-1.5 1.5H5.25a1.5 1.5 0 01-1.5-1.5V8.25a1.5 1.5 0 011.5-1.5h5.25a.75.75 0 000-1.5H5.25z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="tab-content h-full flex"
    >
      <div
        class="w-full h-full px-2 py-2 border-t border-t-gray-300 block fade-in"
      >
        <h3
          class="text-center"
        >
          Imagine you've entered your card data!
        </h3>
        <button
          class="cursor-not-allowed opacity-50 active:scale-100 bg-green-600 text-white font-bold uppercase bg-primary hover:bg-hover hover:shadow-lg transition transform active:scale-90 duration-200 ease-in w-full px-4 py-1.5 flex justify-center items-center gap-2 mt-10"
          disabled=""
        >
          Pay & Go
          <svg
            aria-hidden="true"
            class="w-8 h-8"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.97 3.97a.75.75 0 011.06 0l7.5 7.5a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 11-1.06-1.06l6.22-6.22H3a.75.75 0 010-1.5h16.19l-6.22-6.22a.75.75 0 010-1.06z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        class="w-full h-full px-2 py-2 border-t border-t-gray-300 hidden fade-out"
      >
        <div
          class="max-w-xs h-auto mx-auto shadow-md"
        >
          <div
            class="cepeipt-container px-4 space-y-6 my-6 pb-6 border border-gray-300 rounded-md"
          >
            <div
              class="py-2"
            >
              <h3
                class="text-center text-slate-900 text-3xl"
              >
                Fake App
              </h3>
              <p
                class="text-center text-sm text-gray-400"
              >
                17 December, 2023
              </p>
            </div>
            <div
              class=""
            />
            <div
              class="w-full h-px m-auto bg-gradient-to-r from-transparent via-gray-400 to-transparent"
            />
            <p
              class="receipt-item py-1"
            >
              <span
                class="text-slate-900 text-sm font-bold bg-white px-1 z-10"
              >
                Total
              </span>
              <span
                class="text-slate-900 text-sm font-bold bg-white px-1 z-10"
              >
                â‚¬NaN
              </span>
            </p>
            <h3
              class="text-center text-slate-900 text-2xl"
            >
              Thank you!
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
